Tests for nbastats added

vanandsh opened the pull request:
- added tests for nbastats in tests/testthat/test-nbastats.R
- modified docstring in R/nbastats.R
- man files updated after running document()

vanandsh committed:
Update nbastats.R with functionality

vanandsh committed:
Update test-nbastats for dummy testing

vanandsh committed:
Update nbastats.R warnings to stop

vanandsh committed:
update test-nbastats.R to add more tests for errors

vanandsh committed:
Modify tests for 100% coverage of nbastats.R

vanandsh committed:
Update docstrings for nbastats.R

vanandsh committed:
Run document()

kfoofw commented on R/nbastats.R:
```diff
@@ -3,24 +3,72 @@
+    }
+
+    if (length(positions) != 0) {
+      nba_data <- nba_data[nba_data$POS %in% positions, ]
```
Hi @vanandsh, as discussed verbally, the current `nbastats` function use may be complicated for the user. l

Let's modify the function to `nba_team_stats` where it no longer has a filter input based on player name. This new variant should be able to create descriptive statistics based on different teams. If any positions are specified as input, the filter should return descriptive stats for groupings of team AND positions.

Thanks for your hard work!

vanandsh replied:
Hi @kfoofw 

Thanks for the detailed review. I have made the changes requested for and tested it successfully on my local. Please review the latest changes.

Thanks
Anand

kfoofw commented on tests/testthat/test-nbastats.R:
```diff
@@ -1,3 +1,53 @@
+}
+
+data = read_csv("https://raw.githubusercontent.com/kfoofw/nba_espn/master/0.data/NBA_reg_2018-2019.csv")
+test_nbastats(data)
```
As discussed, please also modify the test scripts accordingly to the new variant of the function.

vanandsh replied:
@kfoofw  The test scripts have also been modified, and coverage is 100%.

kfoofw reviewed (changes requested):
preliminary review of `nbastats`

vanandsh committed:
Remove
- nbastats.R
- nbastats.Rd
- tests/testthat/test-nbastats.R

Files have been renamed to maintain consistency with other files.

vanandsh committed:
Add nba_team_stats.R
- function is replicated to have same functionality like nbastats.R
- all parameter has been removed for overriding other params
- change in argument names
- grouping by Team and (Team, POS) based on team has been introduced.

vanandsh committed:
document() for nba_team_stats.R

vanandsh committed:
test file for nba_team_stats -> test-nba_team_stats.R
- coverage 100%

vanandsh committed:
update NAMESPACE after filename changes

kfoofw commented on tests/testthat/test-nba_team_stats.R:
```diff
@@ -0,0 +1,67 @@
+#'
+#' @examples
+#' test_nbastats(nba_data)
+test_nbastats <- function(nba_data) {
```
```suggestion
test_nba_team_stats<- function(nba_data) {
```

kfoofw commented on tests/testthat/test-nba_team_stats.R:
```diff
@@ -0,0 +1,67 @@
+  })
+}
+nba_data = read_csv("https://raw.githubusercontent.com/kfoofw/nba_espn/master/0.data/NBA_reg_2018-2019.csv")
+test_nbastats(nba_data)
```
```suggestion
test_nba_team_stats(nba_data)
```

kfoofw reviewed (changes requested):
thanks @vanandsh for your edits! Some minor adjustments required. Can you also edit the main repo ReadMe.rmd (and md file) to change the function from `nbastats` to `nba_team_stats`, and the corresponding function description as well? Thanks!

vanandsh committed:
EDIT README to list revised function names and description

vanandsh committed:
Change function name for consistency with file name.

kfoofw committed:
Merge branch 'master' into final

kfoofw reviewed (approved):
thanks for the changes!
